db/pebble/db.go

```diff
@@ -108,3 +117,32 @@ func (d *DB) Update(fn func(txn db.Transaction) error) error {
 func (d *DB) Impl() any {
 	return d.pebble
 }
+
+func CalculatePrefixSize(ctx context.Context, pDB *DB, prefix []byte) (uint, error) {
+	var (
+		err  error
+		size uint
+		v    []byte
+	)
+
+	const upperBoundofPrefix = 0xff
+	pebbleDB := pDB.Impl().(*pebble.DB)
+	it, err := pebbleDB.NewIter(&pebble.IterOptions{LowerBound: prefix, UpperBound: append(prefix, upperBoundofPrefix)})
+	if err != nil {
+		// No need to call utils.RunAndWrapOnError() size iterator couldn't be created
```

Comment:
Did you mean `since` instead of `size` ?

Reply:
Yes. fixed